cluster/gce/util.sh

```diff
@@ -2835,7 +2837,7 @@ function detach-internal-master-ip() {
   # Detach ${MASTER_NAME}-internal-ip from ${name}
   gcloud compute instances network-interfaces update "${name}" --project "${PROJECT}" --zone "${zone}" --aliases="${aliases}"
   gcloud compute instances remove-metadata "${name}" --zone "${zone}" --keys=kube-master-internal-ip
-  run-gcloud-command "${name}" "${zone}" 'sudo ip route del to local '${ip}'/32 dev $(ip route | grep default | awk '\''{print $5}'\'')' || true
+  run-gcloud-command "${name}" "${zone}" "'sudo ip route del to local ${ip}/32 dev \$(ip route | grep default | while read -r d d d d d; do echo \$d; done)'" || true
```

Comment:
This also feels weirder than the awk version to me, what do others think?

Reply:
It is in fact pretty much a schoolbook use of `read`, IMO:
Read line-by-line, divide the 5 first entries by IFS to the variables `a b c d e` where `a-d` are of no interest to us since we want the 5th column only, make shellcheck happy by not having unused variables and use one variable (here: `d`) for all of them and it will contain the last value (5th column) for each line.

Reply:
I have now updated this to use the perhaps less confusing

`while read -r _ _ _ _ dev; do echo \$dev; done`

instead.

Reply:
I think the underscore version might click more readily 👍

Reply:
Now updated to

`while read -r _ _ _ _ dev _`

:-)